src/twisted/mail/test/test_pop3.py

```diff
@@ -382,51 +427,56 @@ def testAuthListing(self):
         d = loopback.loopbackAsync(p, client)
         return d.addCallback(self._cbTestAuthListing, client)
 
+
     def _cbTestAuthListing(self, ignored, client):
         self.assertTrue(client.response[1].startswith('+OK'))
         self.assertEqual(sorted(client.response[2:5]),
                          ["AUTH1", "AUTHLAST", "SECONDAUTH"])
         self.assertEqual(client.response[5], ".")
 
+
     def testIllegalPASS(self):
-        dummy = DummyPOP3()
-        client = LineSendingProtocol([
-            "PASS fooz",
-            "QUIT"
-        ])
-        d = loopback.loopbackAsync(dummy, client)
-        return d.addCallback(self._cbTestIllegalPASS, client, dummy)
+        return self.runTest(
+            ['PASS fooz',
+             'QUIT'],
+            ['+OK <moshez>',
+             '-ERR USER required before PASS',
+             '+OK '])
 
-    def _cbTestIllegalPASS(self, ignored, client, dummy):
-        expected_output = '+OK <moshez>\r\n-ERR USER required before PASS\r\n+OK \r\n'
-        self.assertEqual(expected_output, '\r\n'.join(client.response) + '\r\n')
-        dummy.connectionLost(failure.Failure(Exception("Test harness disconnect")))
 
     def testEmptyPASS(self):
-        dummy = DummyPOP3()
-        client = LineSendingProtocol([
-            "PASS ",
-            "QUIT"
-        ])
-        d = loopback.loopbackAsync(dummy, client)
-        return d.addCallback(self._cbTestEmptyPASS, client, dummy)
+        return self.runTest(
+            ['PASS ',
+             'QUIT'],
+            ['+OK <moshez>',
+             '-ERR USER required before PASS',
+             '+OK '])
+
+
+    def testWhiteSpaceInPASS(self):
```

Comment:
Please add a docstring for this new test method and rename it to follow the naming convention.  Also, please add more test coverage for the new functionality.  For example, multiple words separated by a space and multiple consecutive spaces.  Other kinds of whitespace would also be an interesting thing to test (I don't think that's broken but having the test coverage would be beneficial).

Also, a positive-path test where passwords with spaces in them lead to an authentication _success_ would be good.  Otherwise the test suite just shows it isn't a parse error - it doesn't show it actually works.